### PR TITLE
fix(UI) Fix multiple UI usability issues

### PR DESCRIPTION
**Pull request by chriscollins3456**
Fixes four different usability issues we've heard about in the UI. This PR might be easiest to review by commit as each one fixes its own issue. One of the four issues is broken into two commits (same issue, different places). 

Here are the issues and fixes broken down by commit:

1. Limit number of users for the "Top Users" on entity profiles. Previously we just showed all of the users that interacted with a dataset, but now we pass in a prop to limit the number of users we show, and for this situation let's limit it to 10 (who needs to see their top 43 users?)
2. Users were experiencing issues with the search bar being too small as important parts of their search results were getting cut off in the drop down. So commit 2 simply increases the size of the search bar on the home page.
3. Same issue, but people were experiencing issues on searchable pages with the search bar at the top of the page. Since we have so many admin links on the top header with the search bar, they limit space a decent amount. So now if the search bar is not able to expand wide enough, hide those buttons when they click in the search bar and expand the search bar. Then when they leave the search bar, it will contract back to its normal width and we will show the links again.
4. The search panel was getting very wild and unwieldy. This simplifies things a bit by sorting the search options to have the most important ones at the top. Also, default collapse all the search filters of filter types that are not deemed as important.
5. People were experiencing issues with the lineage viewer if they click to expand their view too quickly between expansions. The issue here was that we were using a single lazy query at the top level for each entity box, and if one query hadn't resolved yet and we made another request, it would cancel the first query and leave the users hanging in a loading state. To fix this, I moved the query down lower into each entity instance and then pass the data back up to construct the tree. Now these queries don't cancel each other out.

The updated side bar for search filters:
image

The top search bar interaction when the screen is not very wide:
ezgif-4-1330b83a0a

## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: datahub-web-react/src/app/search/SearchBar.tsx [hunk 1]
```diff
@@ -161,6 +161,7 @@ interface Props {
     autoCompleteStyle?: React.CSSProperties;
     entityRegistry: EntityRegistry;
     fixAutoComplete?: boolean;
+    setIsInSearchBar?: (isInSearchBar: boolean) => void;
```
Comment (gabe-lyons) [inline datahub-web-react/src/app/search/SearchBar.tsx:164]: lets call this `setIsSearchBarFocused` to be specific?

File: datahub-web-react/src/app/lineage/LineageTreeNodeAndEdgeRenderer.tsx [hunk 2]
```diff
@@ -15,7 +15,7 @@ type Props = {
     };
     onEntityClick: (EntitySelectParams) => void;
     onEntityCenter: (EntitySelectParams) => void;
-    onLineageExpand: (LineageExpandParams) => void;
+    onLineageExpand: (daat: EntityAndType) => void;
```
Comment (gabe-lyons) [inline datahub-web-react/src/app/lineage/LineageTreeNodeAndEdgeRenderer.tsx:18]: typo :)

File: datahub-web-react/src/app/search/SearchBar.tsx (hunk 1 above)
Comment (jjoyce0510) [reply datahub-web-react/src/app/search/SearchBar.tsx:164]: ++

File: datahub-web-react/src/app/lineage/LineageEntityNode.tsx [hunk 3]
```diff
@@ -81,13 +82,20 @@ export default function LineageEntityNode({
...
     const { expandTitles } = useContext(LineageExplorerContext);
     const [isExpanding, setIsExpanding] = useState(false);
     const [expandHover, setExpandHover] = useState(false);
+    const { getAsyncEntity, asyncData } = useLazyGetEntityQuery();
+
+    useEffect(() => {
+        if (asyncData) {
+            onExpandClick(asyncData);
+        }
+    }, [asyncData, onExpandClick]);
```
Comment (gabe-lyons) [inline datahub-web-react/src/app/lineage/LineageEntityNode.tsx:98]: nice! this is a surprisingly small change

**Review by jjoyce0510 — APPROVED** [review]
Overall the LGTM. Nice work, Chris!

File: datahub-web-react/src/app/lineage/LineageTreeNodeAndEdgeRenderer.tsx (hunk 2 above)
Comment (chriscollins3456) [reply datahub-web-react/src/app/lineage/LineageTreeNodeAndEdgeRenderer.tsx:18]: daat!

File: datahub-web-react/src/app/lineage/LineageEntityNode.tsx (hunk 3 above)
Comment (chriscollins3456) [reply datahub-web-react/src/app/lineage/LineageEntityNode.tsx:98]: yeah I was pleasantly surprised!

**Review by gabe-lyons — APPROVED** [review]

**Pull request merged**